pkg/registry/generic/registry/store.go

```diff
@@ -268,6 +306,34 @@ func (e *Store) Update(ctx api.Context, obj runtime.Object) (runtime.Object, boo
 		UIDCopy := meta.UID
 		preconditions = &storage.Preconditions{UID: &UIDCopy}
 	}
+	// check if the update is removing all the object's finalizers. If so,
+	// further checks if the object's DeletionGracePeriodSeconds is 0. If so,
+	// delete the object.
+	existing := e.NewFunc()
+	delete := e.shouldDelete(ctx, key, obj, existing)
```

Comment:
@lavalamp @liggitt because `obj` isn't available after your change, how can I do the check here?